src/server/auth/utils.js

```diff
@@ -27,10 +27,39 @@ export const getAccessibleResourcesFromArboristasync = async (jwt) => {
     }
     throw new CodedError(data.error.code, data.error.message);
   }
+
+  data = resourcePathsWithServiceMethodCombination(data, ['guppy', '*'], ['read', '*'])
   const resources = data.resources ? _.uniq(data.resources) : [];
   return resources;
 };
 
+export const canRefresh = async (jwt) => {
+  let data;
+  if (config.internalLocalTest) {
+    data = {
+      resources: [ // these are just for testing
+        '/programs/DEV/projects/test',
+        '/programs/jnkns/projects/jenkins',
+      ],
+    };
+  } else {
+    data = await arboristClient.listAuthMapping(jwt);
```

Comment:
Since this `canRefresh()` will only be called within `initialize()`, you don't have to make another call to arborist to get user's ath mapping again because you have already done it in `getAccessibleResourcesFromArboristasync()`
consider to put the auth mapping in a vairable and use it here instead of making another call

Reply:
This has been implemented in recent commits